File: server/src/Mentora.Application/DTOs/StudentClassesResponse.cs
namespace Mentora.Application.DTOs;

public record StudentClassesResponse
{
    public Guid ClassId { get; init; }
    public string ClassName { get; init; } = string.Empty;
    public DateTime DateStart { get; init; }
    public DateTime DateEnd { get; init; }
    public bool ClassActive { get; init; }
    public CourseDetail Course { get; init; } = null!;
}

public record CourseDetail
{
    public Guid Id { get; init; }
    public string Name { get; init; } = string.Empty;
    public string FaceImage { get; init; } = string.Empty;
    public int WorkloadHours { get; init; }
    public bool Active { get; init; }
    public CategoryDetail Category { get; init; } = null!;
    public IEnumerable<SlideDetail> Slides { get; init; } = [];
}

public record CategoryDetail
{
    public Guid Id { get; init; }
    public string Name { get; init; } = string.Empty;
}

public record SlideDetail
{
    public Guid Id { get; init; }
    public string Title { get; init; } = string.Empty;
    public string Content { get; init; } = string.Empty;
    public string SlideTypeName { get; init; } = string.Empty;
    public int Ordering { get; init; }
    public bool Active { get; init; }
}

File: mentora-web/src/app/services/responses/student-classes.response.ts
export interface StudentClassesResponse {
  classId: string;
  className: string;
  dateStart: string;
  dateEnd: string;
  classActive: boolean;
  course: CourseDetail;
}

export interface CourseDetail {
  id: string;
  name: string;
  faceImage: string;
  workloadHours: number;
  active: boolean;
  category: CategoryDetail;
  slides: SlideDetail[];
}

export interface CategoryDetail {
  id: string;
  name: string;
}

export interface SlideDetail {
  id: string;
  title: string;
  content: string;
  slideTypeName: string;
  ordering: number;
  active: boolean;
}
